app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -248,6 +251,10 @@ export class CustomOAuth {
 				if (!identity.email && (identity.emails && Array.isArray(identity.emails) && identity.emails.length >= 1)) {
 					identity.email = identity.emails[0].address ? identity.emails[0].address : undefined;
 				}
+				if (this.mergeRoles) {
+					const user = Users.findOneByEmailAddress(identity.email);
+					mapRolesFromSSO(user, identity, this.rolesClaim || 'roles');
+				}
```

Comment:
Also sorry just occured to me.. what happens here if this is a new user registering?  

Maybe this should move inside: `addHookToProcessUser`  after it finds the user record.  Then in validateNewUser could do: `users.roles = ['your', 'roles']` so it will create with those roles